src/components/ProfitAndLossReport/ProfitAndLossReport.tsx

```diff
@@ -23,6 +30,36 @@ export const ProfitAndLossReport = ({
   view,
 }: ProfitAndLossReportProps) => {
   const { comparisonConfig } = useContext(ProfitAndLoss.ComparisonContext)
+  const [selectedLineItem, setSelectedLineItem] = useState<SelectedLineItem | null>(null)
+
+  // Memoize breadcrumb index lookup for O(1) performance
+  const breadcrumbIndexMap = useMemo(() => {
+    if (!selectedLineItem) return new Map<string, number>()
+
+    return new Map(
+      selectedLineItem.breadcrumbPath.map((item, index) => [item.name, index]),
+    )
```

Comment:
Memoization is great! You may not need a `Map` here as compared to a plain JavaScript object, since it does add extra overhead for low numbers of keys (assuming there's <100 keys).
```suggestion
    return selectedLineItem.breadcrumbPath.map((item, index) => [item.name, index])
```